File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:
    def encode(self, strs: List[str]) -> str:
        s=""
        for i in strs:
            s+=i+"|#"
        return s
    def decode(self, s: str) -> List[str]:
        z=s.split("|#")
        return z[:-1]

        

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> mapy;
        vector<vector<string>> out;
        for(int i=0;i<strs.size();i++){
            string temp=strs[i];
            sort(temp.begin(),temp.end());
            mapy[temp].push_back(strs[i]);
        }
        for(auto it:mapy){
            out.push_back(it.second);
        }
        return out;
    }
};
